### PR TITLE
[Metricbeat] Migrate EnvoyProxy server to use ReporterV2 interface

### PR DESCRIPTION
**Pull request by berfinsari**
Refer to elastic/beats#10774 for more info

### CODE REVIEW

**Comment by sayden** [comment]
jenkins, test this please

File: metricbeat/module/envoyproxy/server/server_integration_test.go [hunk 1]
```diff
@@ -29,27 +29,32 @@ import (
...
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
 	}
+	assert.NotEmpty(t, events)
+
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), events[0])
 }
 
-func TestFetch(t *testing.T) {
+func TestData(t *testing.T) {
```
Comment (sayden) [inline metricbeat/module/envoyproxy/server/server_integration_test.go]: As commented in your other PR's, now `TestData` must look like this (sorry for the misleading information at the beginning):

```go
func TestData(t *testing.T) {
	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())

	if err := mbtest.WriteEventsReporterV2Error(f, t, ""); err != nil {
		t.Fatal("write", err)
	}
}
```

Comment (ruflin) [reply metricbeat/module/envoyproxy/server/server_integration_test.go]: The above code is for the new error interface and will not work here.

Comment (sayden) [reply metricbeat/module/envoyproxy/server/server_integration_test.go]: Updated, in any case, better to merge it like this and do a follow up PR.

**Comment by sayden** [comment]
Thank you very much for your contribution @berfinsari !

**Pull request merged**